beancount.el

```diff
@@ -763,21 +767,26 @@ what that column is and returns it (an integer)."
         ;; The account has declared a single currency, so we can fill it in.
         (match-string-no-properties 1)))))
 
-(defun beancount--electric-currency ()
-  (when (and beancount-electric-currency (eq last-command-event ?\n))
+(defun beancount--electric ()
+  (when (and (eq last-command-event ?\n)
+             (or beancount-electric-currency
+                 beancount-electric-align))
     (save-excursion
       (forward-line -1)
-      (when (and (beancount-inside-transaction-p)
-                 (looking-at (concat "\\s-+\\(" beancount-account-regexp "\\)"
-                                     "\\s-+\\(" beancount-number-regexp "\\)\\s-*$")))
-        ;; Last line is a posting without currency.
-        (let* ((account (match-string 1))
-               (pos (match-end 0))
-               (currency (beancount--account-currency account)))
-          (when currency
-            (save-excursion
-	      (goto-char pos)
-              (insert " " currency))))))))
+      (when (beancount-inside-transaction-p)
+        (when (and beancount-electric-currency
+                   (looking-at (concat "\\s-+\\(" beancount-account-regexp "\\)"
+                                       "\\s-+\\(" beancount-number-regexp "\\)\\s-*$")))
+          ;; Last line is a posting without currency.
+          (let* ((account (match-string 1))
+                 (pos (match-end 0))
+                 (currency (beancount--account-currency account)))
+            (when currency
+              (save-excursion
+                (goto-char pos)
+                (insert " " currency)))))
+        (when beancount-electric-align
+          (beancount-align-number (beancount-number-alignment-column)))))))
```

Comment:
I don't think this is the right way to implement it. The behavior you want should be implemented in the indent function which now has an extra conditional that should be removed... Well, it is easier if I finally push my local changes than for me to explain where the problem is. Please give me a little time to do that.

Reply:
I just pushed a commit that should fix this. Please let me know if this now behaves as you expect. It does for me. Sorry if I two lines patch took forever to materialize.

Reply:
This works perfectly! Thanks for your help (and for this package).